db/db_impl/db_impl_write.cc

```diff
@@ -2048,7 +2048,7 @@ Status DBImpl::ScheduleFlushes(WriteContext* context) {
 
   MemTableSwitchRecord mem_switch_record;
   std::string replication_sequence;
-  if (immutable_db_options_.replication_log_listener) {
+  if (immutable_db_options_.replication_log_listener && !cfds.empty()) {
```

Comment:
theoretically, `cfds` can't be empty for auto flush. But it doesn't hurt to check.